Edit sampling site fixes

mauberti-bc opened the pull request:
## Links to Jira Tickets

- N/A
- Fixes the issue where stratums and blocks cannot be added to a sampling site when editing the site

## Description of Changes

- When a sampling site is edited, the edit page now requests data for that specific sampling site to initialize the formik values
- Added .kml support for sampling site location file uploads
- Updated tests for sample-location-repository and the sample location GET endpoint
- ```additionalProperties: false``` is added to the blocks and stratums objects of the sampling site PUT endpoint
- Renamed ```methods``` and ```periods``` properties to ```sample_methods``` and ```sample_periods``` to help reassign types
- Removed references to ```surveyed_all_areas```, a deprecated field
- Rearranged survey form files into subdirectories

## Testing Notes

- Should be able to create and edit a sampling site with blocks and stratums
- Should be able to create and edit sampling sites with .kml

NickPhura committed:
Initial working observation import against sampling period

NickPhura committed:
Add better loading/disabled handlers

NickPhura committed:
Merge remote-tracking branch 'origin/dev' into SIMSBIOHUB-501

NickPhura committed:
Add Knip and SWC

NickPhura committed:
Update import observations

NickPhura committed:
Remove console logs

NickPhura committed:
Remove swc from API to fix unit tests

NickPhura committed:
Remove knip

NickPhura committed:
Remove gulp

NickPhura committed:
Update lock

NickPhura committed:
Merge remote-tracking branch 'origin/dev' into SIMSBIOHUB-501

mauberti-bc committed:
standards page and api service

NickPhura committed:
Merge branch 'dev' into SIMSBIOHUB-501

NickPhura committed:
Merge branch 'SIMSBIOHUB-501' of https://github.com/bcgov/biohubbc into SIMSBIOHUB-501

NickPhura committed:
Add unit tests. fix spelling

NickPhura committed:
Add tests

mauberti-bc committed:
update standards page

mauberti-bc committed:
standards endpoint and service tests

mauberti-bc committed:
finish unit tests

mauberti-bc committed:
linter

mauberti-bc committed:
add standards to hamburger menu for smaller screens

mauberti-bc committed:
fix openapi spec

mauberti-bc committed:
linter

NickPhura committed:
Fix prettier (bump eslint/prettier versions)

NickPhura committed:
Update .eslintrc

NickPhura committed:
Update .eslintrc
Fix issues

NickPhura committed:
re-run install

NickPhura committed:
Merge branch 'dev' into npTechDebt

mauberti-bc committed:
Merge remote-tracking branch 'origin/npTechDebt' into taxon-standards

mauberti-bc committed:
prettier changes & node update

mauberti-bc committed:
Merge branch 'taxon-standards' of github.com:bcgov/biohubbc into taxon-standards

mauberti-bc committed:
linter

mauberti-bc committed:
Merge branch 'dev' of github.com:bcgov/biohubbc into taxon-standards

mauberti-bc committed:
attempt to solve prettier issues

mauberti-bc committed:
merge in dev

mauberti-bc committed:
basic configure table modal

mauberti-bc committed:
update file structure

mauberti-bc committed:
refactor sampling site page

mauberti-bc committed:
fix measurement issue

mauberti-bc committed:
merge in dev

mauberti-bc committed:
organize sample site create and edit flows

mauberti-bc committed:
linter

mauberti-bc committed:
linter

mauberti-bc committed:
fix tests

mauberti-bc committed:
fix formik validation

mauberti-bc committed:
fix edit survey types

mauberti-bc committed:
cleanup

mauberti-bc committed:
fix test

mauberti-bc committed:
jsdoc and cleanup

mauberti-bc committed:
missing dependency

mauberti-bc committed:
fix duplicate keys & code smells

mauberti-bc committed:
refactor survey folder subdirectories

mauberti-bc committed:
prettier

mauberti-bc committed:
fix file paths in tests

mauberti-bc committed:
merge in dev

mauberti-bc committed:
replace sample_blocks and sample_stratums with blocks and stratums, respectively

mauberti-bc committed:
remove extra fields from project details response

mauberti-bc committed:
prettier

mauberti-bc committed:
re-remove surveyed_areas_all

mauberti-bc committed:
prettier

mauberti-bc committed:
merge in remote

mauberti-bc committed:
update openapi sec

mauberti-bc committed:
update tests

mauberti-bc committed:
fix sonarcloud alert

mauberti-bc committed:
additionalproperties: false

NickPhura commented:
Getting an error when I try and load the survey page:

(In this case, I'm loading a default survey generated by the seed)

```
{
  timestamp: '2024-05-01 14:07:55',
  level: 'error',
  logger: 'paths/project/{projectId}/survey/{surveyId}/view',
  label: 'getSurveyForView',
  message: 'error',
  error: Error executing SQL query: SQL response failed schema check
      at new ApiError (/opt/app-root/src/src/errors/api-error.ts:23:13)
      at new ApiExecuteSQLError (/opt/app-root/src/src/errors/api-error.ts:67:5)
      at parseError (/opt/app-root/src/src/database/db-utils.ts:67:11)
      at Object.sql (/opt/app-root/src/src/database/db-utils.ts:39:13)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async SurveyLocationRepository.getSurveyLocationsData (/opt/app-root/src/src/repositories/survey-location-repository.ts:99:22)
      at async SurveyService.getSurveyById (/opt/app-root/src/src/services/survey-service.ts:83:18)
      at async /opt/app-root/src/src/paths/project/{projectId}/survey/{surveyId}/view.ts:167:26 {
    errors: [
      ZodError: [
        {
          "code": "invalid_type",
          "expected": "object",
          "received": "undefined",
          "path": [
            "rows",
            0,
            "geometry"
          ],
          "message": "Required"
        },
        {
          "code": "invalid_type",
          "expected": "string",
          "received": "undefined",
          "path": [
            "rows",
            0,
            "geography"
          ],
          "message": "Required"
        }
      ]
          at Object.get error [as error] (/opt/app-root/src/node_modules/zod/lib/types.js:55:31)
          at ZodObject.parseAsync (/opt/app-root/src/node_modules/zod/lib/types.js:183:22)
          at processTicksAndRejections (node:internal/process/task_queues:95:5)
          at async Object.sql (/opt/app-root/src/src/database/db-utils.ts:37:14)
          at async SurveyLocationRepository.getSurveyLocationsData (/opt/app-root/src/src/repositories/survey-location-repository.ts:99:22)
          at async SurveyService.getSurveyById (/opt/app-root/src/src/services/survey-service.ts:83:18)
          at async /opt/app-root/src/src/paths/project/{projectId}/survey/{surveyId}/view.ts:167:26 {
        issues: [
          {
            code: 'invalid_type',
            expected: 'object',
            received: 'undefined',
            path: [Array],
            message: 'Required'
          },
          {
            code: 'invalid_type',
            expected: 'string',
            received: 'undefined',
            path: [Array],
            message: 'Required'
          }
        ],
        addIssue: [Function (anonymous)],
        addIssues: [Function (anonymous)],
        errors: [
          {
            code: 'invalid_type',
            expected: 'object',
            received: 'undefined',
            path: [Array],
            message: 'Required'
          },
          {
            code: 'invalid_type',
            expected: 'string',
            received: 'undefined',
            path: [Array],
            message: 'Required'
          }
        ]
      }
    ]
  }
}
```

NickPhura committed:
Merge remote-tracking branch 'origin/dev' into sample-site-req

mauberti-bc committed:
kml support

mauberti-bc commented:
> Getting an error when I try and load the survey page:
> 
> (In this case, I'm loading a default survey generated by the seed)
> 
> ```
> {
>   timestamp: '2024-05-01 14:07:55',
>   level: 'error',
>   logger: 'paths/project/{projectId}/survey/{surveyId}/view',
>   label: 'getSurveyForView',
>   message: 'error',
>   error: Error executing SQL query: SQL response failed schema check
>       at new ApiError (/opt/app-root/src/src/errors/api-error.ts:23:13)
>       at new ApiExecuteSQLError (/opt/app-root/src/src/errors/api-error.ts:67:5)
>       at parseError (/opt/app-root/src/src/database/db-utils.ts:67:11)
>       at Object.sql (/opt/app-root/src/src/database/db-utils.ts:39:13)
>       at processTicksAndRejections (node:internal/process/task_queues:95:5)
>       at async SurveyLocationRepository.getSurveyLocationsData (/opt/app-root/src/src/repositories/survey-location-repository.ts:99:22)
>       at async SurveyService.getSurveyById (/opt/app-root/src/src/services/survey-service.ts:83:18)
>       at async /opt/app-root/src/src/paths/project/{projectId}/survey/{surveyId}/view.ts:167:26 {
>     errors: [
>       ZodError: [
>         {
>           "code": "invalid_type",
>           "expected": "object",
>           "received": "undefined",
>           "path": [
>             "rows",
>             0,
>             "geometry"
>           ],
>           "message": "Required"
>         },
>         {
>           "code": "invalid_type",
>           "expected": "string",
>           "received": "undefined",
>           "path": [
>             "rows",
>             0,
>             "geography"
>           ],
>           "message": "Required"
>         }
>       ]
>           at Object.get error [as error] (/opt/app-root/src/node_modules/zod/lib/types.js:55:31)
>           at ZodObject.parseAsync (/opt/app-root/src/node_modules/zod/lib/types.js:183:22)
>           at processTicksAndRejections (node:internal/process/task_queues:95:5)
>           at async Object.sql (/opt/app-root/src/src/database/db-utils.ts:37:14)
>           at async SurveyLocationRepository.getSurveyLocationsData (/opt/app-root/src/src/repositories/survey-location-repository.ts:99:22)
>           at async SurveyService.getSurveyById (/opt/app-root/src/src/services/survey-service.ts:83:18)
>           at async /opt/app-root/src/src/paths/project/{projectId}/survey/{surveyId}/view.ts:167:26 {
>         issues: [
>           {
>             code: 'invalid_type',
>             expected: 'object',
>             received: 'undefined',
>             path: [Array],
>             message: 'Required'
>           },
>           {
>             code: 'invalid_type',
>             expected: 'string',
>             received: 'undefined',
>             path: [Array],
>             message: 'Required'
>           }
>         ],
>         addIssue: [Function (anonymous)],
>         addIssues: [Function (anonymous)],
>         errors: [
>           {
>             code: 'invalid_type',
>             expected: 'object',
>             received: 'undefined',
>             path: [Array],
>             message: 'Required'
>           },
>           {
>             code: 'invalid_type',
>             expected: 'string',
>             received: 'undefined',
>             path: [Array],
>             message: 'Required'
>           }
>         ]
>       }
>     ]
>   }
> }
> ```

Fixed

NickPhura committed:
Refactor database zod schema validation. Now strict.
Fix various endpoints and type definitions to match actual responses.
Misc tweaks/updates/cleanup.

NickPhura committed:
Fix strict zod issues.

NickPhura committed:
Fix test
